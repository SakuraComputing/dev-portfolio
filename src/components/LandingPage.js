import React from 'react';
import Header from '../components/Header';
import About from '../components/About';
import Project from './projects/Project';
import OtherProjects from './projects/OtherProjects';
import Experience from './experience/Experience';
import InfoAside from '../components/InfoAside';
import SkillsAside from './skills/SkillsAside';
import EducationAside from './education/EducationAside';
import LanguagesAside from './languages/LanguagesAside';
import BlogAside from '../components/BlogAside';
import MusicAside from './music/MusicAside';
import CreditsAside from '../components/CreditsAside';
import Footer from '../components/Footer';
import profile_json from '../../data/profile';

class LandingPage  extends React.Component {

    constructor(props) {
        super(props);
         this.state = {};
    }


    componentWillMount() {
        this.setState(profile_json);
    }

    render() {
        const { header,
                about,
                infoaside,
                skillsaside,
                skillsummary,
                musicaside,
                projects,
                latestproject,
                experience,
                otherprojects,
                education,
                languages
        } = this.state;

        return (
            <div>
                <Header header={header}/>
                <div className="container sections-wrapper">
                    <div className="row">
                        <div className="primary col-md-8 col-sm-12 col-xs-12">
                            <About about={about} />
                            <Project projects={projects} latestproject={latestproject}/>
                            <OtherProjects otherprojects={otherprojects}/>
                            <Experience experience={experience}/>
                        </div>
                        <div className="secondary col-md-4 col-sm-12 col-xs-12">
                            <InfoAside infoaside={infoaside}/>
                            <SkillsAside skillsaside={skillsaside} skillsummary={skillsummary}/>
                            <EducationAside education={education}/>
                            <LanguagesAside languages={languages}/>
                            <BlogAside/>
                            <MusicAside musicaside={musicaside}/>
                            <CreditsAside/>
                        </div>
                    </div>
                </div>
                <Footer/>
            </div>
        )
    }
};
export default LandingPage;
