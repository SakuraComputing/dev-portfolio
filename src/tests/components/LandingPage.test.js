import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import LandingPage from '../../components/LandingPage';
import profile_header_json from '../../../data/profile';


describe('Landing Page', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<LandingPage />);
    });

    it('should render Expense dashboard page correctly',() => {
        expect(toJSON(wrapper)).toMatchSnapshot();
    });
    it('should read the profile json file into state', () => {

        // const profileState = {
        //     name: 'test'
        // };
        // wrapper.setState(profileState);
        expect(wrapper.state()).toEqual(profile_header_json);
    });
});
