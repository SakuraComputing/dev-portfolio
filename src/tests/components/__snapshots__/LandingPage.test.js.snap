// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing Page should render Expense dashboard page correctly 1`] = `
<div>
  <Header
    header={
      Object {
        "name": "Ewan Bauld",
        "social": Object {
          "github": Object {
            "url": "https://github.com/SakuraComputing",
          },
          "googleplus": Object {
            "url": "",
          },
          "hackernews": Object {
            "url": "",
          },
          "linkedin": Object {
            "url": "https://www.linkedin.com/in/ewan-bauld-8887a337/",
          },
          "twitter": Object {
            "url": "",
          },
        },
        "title": "Senior Web Developer",
      }
    }
  />
  <div
    className="container sections-wrapper"
  >
    <div
      className="row"
    >
      <div
        className="primary col-md-8 col-sm-12 col-xs-12"
      >
        <About
          about={
            Object {
              "text": "Front end developer with over twnety years of experience in software delivery.  Specialising in ReactJS but also experience in vanilla javascript and other MVC frameworks such as Angular. Take both TDD approaches and AB Testing to coding",
            }
          }
        />
        <LatestProject />
        <OtherProjects />
        <Experience />
        <Github />
      </div>
      <div
        className="secondary col-md-4 col-sm-12 col-xs-12"
      >
        <InfoAside
          infoaside={
            Object {
              "email": "ewan.bauld@gmail.com",
              "location": "Manchester, UK",
              "website": "www.ewan.bauld.com",
            }
          }
        />
        <SkillsAside
          skillsaside={
            Array [
              Object {
                "bar": 70,
                "id": 1,
                "level": "Advanced",
                "title": "CSS",
              },
              Object {
                "bar": 81,
                "id": 2,
                "level": "Expert",
                "title": "Javascript",
              },
            ]
          }
        />
        <TestimonialsAside />
        <EducationAside />
        <LanguagesAside />
        <BlogAside />
        <MusicAside />
        <ConferenceAside />
        <CreditsAside />
      </div>
    </div>
  </div>
  <Footer />
</div>
`;
